### PR TITLE
fix: improve baseapp event emission (backport cosmos/cosmos-sdk#14356)

### PR DESCRIPTION
**Pull request by 0Tech**
## Description

ref: #1076 

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: baseapp/baseapp.go [hunk 1]
```diff
@@ -847,12 +847,15 @@ func createEvents(msg sdk.Msg) sdk.Events {
 		msgEvent = msgEvent.AppendAttributes(sdk.NewAttribute(sdk.AttributeKeySender, msg.GetSigners()[0].String()))
 	}
 
-	// here we assume that routes module name is the second element of the route
-	// e.g. "cosmos.bank.v1beta1.MsgSend" => "bank"
-	moduleName := strings.Split(eventMsgName, ".")
-	if len(moduleName) > 1 {
-		msgEvent = msgEvent.AppendAttributes(sdk.NewAttribute(sdk.AttributeKeyModule, moduleName[1]))
+	// verify that events have no module attribute set
+	if _, found := events.GetAttributes(sdk.AttributeKeyModule); !found {
```
Comment (zemyblue) [inline baseapp/baseapp.go:851]: Does all tx event of ibc include `module` attribute?

Comment (0Tech) [reply baseapp/baseapp.go:851]: The supported version of `Finschia/ibc-go/v4` is `v4.3.1`. Hence, I will check it on that version.

By the way, this PR would not remove any information included before.

**Review by jaeseung-bae — APPROVED** [review]

File: baseapp/baseapp.go (hunk 1 above)
Comment (0Tech) [reply baseapp/baseapp.go:851]: > Does all tx event of ibc include `module` attribute?

No, the current version, `v4.3.1`, does not include `module` attribute in all its events. Some logic lacks the event.

Comment (zemyblue) [reply baseapp/baseapp.go:851]: If then, does the all ibc events display no ibc module name?

Comment (0Tech) [reply baseapp/baseapp.go:851]: Some logics emit the events, but others don't.

Comment (0Tech) [reply baseapp/baseapp.go:851]: I will add logic not to append attribute `module` in case of `ibc`. It would be addressed by another PR.

**Review by zemyblue — APPROVED** [review]

**Pull request merged**